AK/NumberFormat.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2020-2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "AK/Assertions.h"
+#include "AK/DeprecatedString.h"
+#include <AK/NumberFormat.h>
+#include <AK/StringView.h>
+
+namespace AK {
+
+// FIXME: Remove this hackery once printf() supports floats.
+static DeprecatedString number_string_with_one_decimal(u64 number, u64 unit, StringView suffix)
+{
+    int decimal = (number % unit) * 10 / unit;
+    return DeprecatedString::formatted("{}.{} {}", number / unit, decimal, suffix);
+}
+
+template<size_t Size>
+static DeprecatedString human_readable_quantity_impl(u64 quantity, const AK::Array<AK::StringView, Size>& suffixes)
+{
+    VERIFY(suffixes.size() >= 2);
+
+    u64 size_of_unit = 1024;
+    if (quantity < size_of_unit)
+        return DeprecatedString::formatted("{} {}", quantity, suffixes.first());
+
+    for (size_t i = 1; i < suffixes.size() - 1; i++) {
+        auto suffix = suffixes[i];
+        if (quantity < 1024 * size_of_unit) {
+            return number_string_with_one_decimal(quantity, size_of_unit, suffix);
+        }
+
+        size_of_unit *= 1024;
+    }
+
+    return number_string_with_one_decimal(quantity, size_of_unit, suffixes.last());
+}
+
+DeprecatedString human_readable_size(u64 size)
+{
+    static auto suffixes = AK::Array { "B"sv, "KiB"sv, "MiB"sv, "GiB"sv, "TiB"sv, "PiB"sv, "EiB"sv };
```

Comment:
Making this `constexpr` avoids the need to check if it has been initialized each time the function is called.